### PR TITLE
8294138: [11u] Revert change from JDK-8210962 in basic.m4

### PR DESCRIPTION
**Pull request by RealCLanger**
A part of [JDK-8210962](https://bugs.openjdk.org/browse/JDK-8210962) has been introduced in jdk11u-dev via the backport of [JDK-8239708](https://bugs.openjdk.org/browse/JDK-8239708) with this [commit](https://github.com/openjdk/jdk11u-dev/commit/75b3e45fe9d16a8e29f4fd51f93e01acc9436562#diff-9cc5f1086e5a06c7e88857b7d319eb938a76d4fb28905cb3dda3bb7493e809a7).

This change should be reverted.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294138](https://bugs.openjdk.org/browse/JDK-8294138): [11u] Revert change from JDK-8210962 in basic.m4

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev pull/1368/head:pull/1368` \
`$ git checkout pull/1368`

Update a local copy of the PR: \
`$ git checkout pull/1368` \
`$ git pull https://git.openjdk.org/jdk11u-dev pull/1368/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1368`

View PR using the GUI difftool: \
`$ git pr show -t 1368`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/1368.diff">https://git.openjdk.org/jdk11u-dev/pull/1368.diff</a>

</details>

### CODE REVIEW

**Comment by RealCLanger** [comment]
@shipilev FYA - WDYT?

File: make/autoconf/jdk-options.m4 [hunk 1]
```diff
@@ -33,19 +33,8 @@
...
-  if test "x$with_jdk_variant" = xnormal || test "x$with_jdk_variant" = x; then
-    JDK_VARIANT="normal"
-  else
-    AC_MSG_ERROR([The available JDK variants are: normal])
-  fi
-
-  AC_SUBST(JDK_VARIANT)
-
-  AC_MSG_RESULT([$JDK_VARIANT])
+  # Deprecated in JDK 12
```
Comment (shipilev) [inline make/autoconf/jdk-options.m4:36]: This comment is now misleading.

**Comment by gdams** [comment]
looks good to me, apologies for missing this when I backported [JDK-8239708](https://bugs.openjdk.org/browse/JDK-8239708).

**Review by shipilev — COMMENTED** [review]
At this point I am thinking that reverting this little piece might be cleaner than trying to backport the rest of it. My rationale would be making 11u downstream builders' lives easier a bit. But this is something for 11u maintainers to decide.

**Comment by RealCLanger** [comment]
> At this point I am thinking that reverting this little piece might be cleaner than trying to backport the rest of it. My rationale would be making 11u downstream builders' lives easier a bit. But this is something for 11u maintainers to decide.

My gut feeling tells me the same. We probably don't want to rouse folks unnecessarily with such kind of changes. I'll turn this PR into a revert of the change in basic.m4.

**Comment by RealCLanger** [comment]
> At this point I am thinking that reverting this little piece might be cleaner than trying to backport the rest of it. My rationale would be making 11u downstream builders' lives easier a bit. But this is something for 11u maintainers to decide.

OK, changed this into a revert. Please review.

**Review by shipilev — APPROVED** [review]
The change looks good. I tested 11u-dev now configures itself into `*-normal-*`, as usual.

**Comment by RealCLanger** [comment]
> The change looks good. I tested 11u-dev now configures itself into `*-normal-*`, as usual.

Thanks. Flagged for approval.

**Pull request closed without merging**